nextcord/interactions.py

```diff
@@ -559,6 +539,485 @@ async def send(
             allowed_mentions=allowed_mentions,
         )
 
+
+class ApplicationCommandInteraction(Interaction):
+    """Represents the base class for all Discord interactions.
+
+    .. container:: operations
+
+        .. describe:: x == y
+
+            Checks if two interactions are equal.
+
+        .. describe:: x != y
+
+            Checks if two interactions are not equal.
+
+        .. describe:: hash(x)
+
+            Returns the interaction's hash.
+
+    .. versionadded:: 2.0
+
+    .. versionchanged:: 2.1
+        :class:`Interaction` is now hashable.
+
+    Attributes
+    ----------
+    id: :class:`int`
+        The interaction's ID.
+    type: :class:`InteractionType`
+        The interaction type.
+    guild_id: Optional[:class:`int`]
+        The guild ID the interaction was sent from.
+    channel_id: Optional[:class:`int`]
+        The channel ID the interaction was sent from.
+    locale: Optional[:class:`str`]
+        The users locale.
+    guild_locale: Optional[:class:`str`]
+        The guilds preferred locale, if invoked in a guild.
+    application_id: :class:`int`
+        The application ID that the interaction was for.
+    user: Optional[Union[:class:`User`, :class:`Member`]]
+        The user or member that sent the interaction.
+    token: :class:`str`
+        The token to continue the interaction. These are valid
+        for 15 minutes.
+    data: :class:`dict`
```

Comment:
It would be nice if this `data` attribute was uniquely typed for each interaction subclass to use a corresponding typed dict.

Reply:
Issue is, `nextcord.types` in general isn't documented, would it make more sense to simply link to the discord docs?

Reply:
I suppose.

Reply:
As an side, possibly we should discuss introducing something like `discord-typings` if we want to go that route? @ooliver1

Reply:
`discord-typings` too does not have documentation